plugin-server/src/worker/ingestion/person-state-manager.ts

```diff
@@ -0,0 +1,399 @@
+import { Properties } from '@posthog/plugin-scaffold'
+import * as Sentry from '@sentry/node'
+import equal from 'fast-deep-equal'
+import { StatsD } from 'hot-shots'
+import { ProducerRecord } from 'kafkajs'
+import { DateTime } from 'luxon'
+import { DatabaseError } from 'pg'
+
+import { Person, PropertyUpdateOperation } from '../../types'
+import { DB } from '../../utils/db/db'
+import { status } from '../../utils/status'
+import { UUIDT } from '../../utils/utils'
+import { PersonManager } from './person-manager'
+
+const MAX_FAILED_PERSON_MERGE_ATTEMPTS = 3
+
+// used to prevent identify from being used with generic IDs
+// that we can safely assume stem from a bug or mistake
+const CASE_INSENSITIVE_ILLEGAL_IDS = new Set([
+    'anonymous',
+    'guest',
+    'distinctid',
+    'distinct_id',
+    'id',
+    'not_authenticated',
+    'email',
+    'undefined',
+    'true',
+    'false',
+])
+
+const CASE_SENSITIVE_ILLEGAL_IDS = new Set(['[object Object]', 'NaN', 'None', 'none', 'null', '0'])
+
+const isDistinctIdIllegal = (id: string): boolean => {
+    return id.trim() === '' || CASE_INSENSITIVE_ILLEGAL_IDS.has(id.toLowerCase()) || CASE_SENSITIVE_ILLEGAL_IDS.has(id)
+}
+
+// This class is responsible for creating/updating a single person through the process-event pipeline
+export class PersonStateManager {
+    timestamp: DateTime
+    newUuid: string
```

Comment:
This service is coupled to a single "events" lifecycle (rather than being thread-based as usual) - basically responsible for managing a persons state and updating/creating/etc for that purpose.

I'll be passing (and using) more properties in the next PR.

I drew a blank on a good name though, suggestions? It's not a pure class as `PersonState` would indicate.